src/sidecar/connections.ts

```diff
@@ -322,20 +314,68 @@ async function discoverSchemaRegistry(): Promise<string | undefined> {
  * updates that are not relevant to the caller w/o gaps where there could be 0 observers and therefore
  * missed updates.
  * */
-export type ConnectionUpdatePredicate = (connection: Connection) => boolean;
+export type ConnectionUpdatePredicate = (event: ConnectionEventBody) => boolean;
 
 /** Entry type kept in a per connection-id Map within ConnectionStateWatcher. */
-class SingleConnectionState {
-  mostRecentState: Connection | null = null;
-  eventEmitter: EventEmitter<Connection> = new EventEmitter<Connection>();
+export class SingleConnectionEntry {
+  connectionId: ConnectionId;
+  mostRecentEvent: ConnectionEventBody | null = null;
+  eventEmitter: EventEmitter<ConnectionEventBody> = new EventEmitter<ConnectionEventBody>();
+
+  constructor(connectionId: ConnectionId) {
```

Comment:
Each SingleConnectionEntry, per-connection-id constructed either when we receive the first connection change websocket event from sidecar, or `waitForConnectionToBeStable()` is called, will now always handle any connection change events routed into it by calling `connectionEventHandler`.